src/main.ts

```diff
@@ -138,55 +138,74 @@ const ruleSets: {
-        const validateTypeScriptConfig =
-          (argv['validate-tsconfig'] as TypeScriptConfigValidationRuleSet) ??
-          projectInfo.packageJson.jsii?.validateTsconfig ??
-          TypeScriptConfigValidationRuleSet.STRICT;
-
-        const compiler = new Compiler({
-          projectInfo,
-          projectReferences: argv['project-references'],
-          failOnWarnings: argv['fail-on-warnings'],
-          stripDeprecated: argv['strip-deprecated'] != null,
-          stripDeprecatedAllowListFile: argv['strip-deprecated'],
-          addDeprecationWarnings: argv['add-deprecation-warnings'],
-          generateTypeScriptConfig: argv['generate-tsconfig'],
-          typeScriptConfig,
-          validateTypeScriptConfig,
-          compressAssembly: argv['compress-assembly'],
-        });
-
-        const emitResult = argv.watch ? await compiler.watch() : compiler.emit();
-
-        const allDiagnostics = [...projectInfoDiagnostics, ...emitResult.diagnostics];
-
-        for (const diagnostic of allDiagnostics) {
-          utils.logDiagnostic(diagnostic, projectRoot);
-        }
-        if (emitResult.emitSkipped) {
-          process.exitCode = 1;
+          const typeScriptConfig = argv.tsconfig ?? projectInfo.packageJson.jsii?.tsconfig;
+          const validateTypeScriptConfig =
+            (argv['validate-tsconfig'] as TypeScriptConfigValidationRuleSet) ??
+            projectInfo.packageJson.jsii?.validateTsconfig ??
+            TypeScriptConfigValidationRuleSet.STRICT;
+
+          const compiler = new Compiler({
+            projectInfo,
+            projectReferences: argv['project-references'],
+            failOnWarnings: argv['fail-on-warnings'],
+            stripDeprecated: argv['strip-deprecated'] != null,
+            stripDeprecatedAllowListFile: argv['strip-deprecated'],
+            addDeprecationWarnings: argv['add-deprecation-warnings'],
+            generateTypeScriptConfig: argv['generate-tsconfig'],
+            typeScriptConfig,
+            validateTypeScriptConfig,
+            compressAssembly: argv['compress-assembly'],
+          });
+
+          const emitResult = argv.watch ? await compiler.watch() : compiler.emit();
+
+          const allDiagnostics = [...projectInfoDiagnostics, ...emitResult.diagnostics];
+
+          for (const diagnostic of allDiagnostics) {
+            utils.logDiagnostic(diagnostic, projectRoot);
+          }
+          if (emitResult.emitSkipped) {
+            process.exitCode = 1;
+          }
+        } catch (e: unknown) {
+          if (e instanceof utils.JsiiError) {
+            if (e.showHelp) {
+              console.log();
+              yargs.showHelp();
+              console.log();
+            }
+
+            const LOG = log4js.getLogger(utils.CLI_LOGGER);
+            LOG.error(e.message);
+
+            process.exitCode = -1;
+          } else {
+            throw e;
+          }
```

Comment:
The diff for this file is weird. This is the new part + the code above is now wrapped in a try-catch.